peps/pep-0776.rst

```diff
@@ -0,0 +1,794 @@
+Status: Draft
+Type: Standards Track
+Topic: Packaging
+Created: 18-Mar-2025
+Python-Version: 3.14
+Post-History: `18-Mar-2025 <https://discuss.python.org/t/84996>`__
+
+Abstract
+========
+
+`Emscripten <https://emscripten.org/>`__ is a complete open source compiler
+toolchain which compiles C/C++ code into WebAssembly/JavaScript executables that
+run in JavaScript runtimes including browsers and Node.js.
+
+This PEP formalizes the addition of Tier 3 for Emscripten support in Python 3.14
+which `was approved by the steering council on October 25, 2024
+<https://github.com/python/steering-council/issues/256>`__. The goal is to allow
+Pyodide wheels to be uploaded to PyPI.
+
+Motivation
+==========
+
+A web browser is a universal computing platform, available on Windows, macOS,
+Linux, and every smartphone.
+
+`The Pyodide project <https://pyodide.org/>`__ has for years supported
+Emscripten Python. Hundreds of thousands of students have learned Python through
+Pyodide via projects like `Capytale
+<https://web.archive.org/web/20241211090946/https://cfp.jupytercon.com/2023/talk/TJ9YEV/>`__
+and `PyodideU <https://stanford.edu/~cpiech/bio/papers/pyodideU.pdf>`__. Pyodide
+is also increasingly being used by Python packages to provide interactive
+documentation. This demonstrates both the importance and the maturity of the
+Emscripten platform.
+
+Emscripten and WASI are also the only supported platforms that offer any
+meaningful sandboxing.
+
+Goals
+=====
+
+It is our long term goal to upstream the entire Pyodide runtime into CPython,
+but this is out of scope for the present PEP. This PEP only attempts to
+establish the foundation for future work.
+
+Runtime Goals
+-------------
+
+1. To describe the current state of the CPython Emscripten runtime
+2. To describe the current state of the Pyodide runtime
+3. To identify minor features to be upstreamed from the Pyodide runtime into the
+   CPython Emscripten runtime
+
+The minor features identified here are all features that could be implemented
+without a PEP. We discuss more significant runtime features that we would like
+to implement but we defer decisions on those features to subsequent PEPs.
+
+Packaging Goals
+---------------
+
+1. To describe Pyodide's packaging tooling
+2. To describe Pyodide's wheel abi to a similar level of detail as the manylinux
+   PEPs
+3. To request that Pyodide wheels be allowed for upload to PyPI
+
+
+Emscripten Platform Information
+===============================
+
+Background on Emscripten
+------------------------
+
+`Emscripten
+<https://emscripten.org/docs/introducing_emscripten/about_emscripten.html>`__
+consists of a C and C++ compiler and linker based on LLVM, together with a
+runtime based on a mildly patched musl libc.
+
+Emscripten is a POSIX-based platform. It uses the WebAssembly binary format,
+specified here:
+
+* https://webassembly.github.io/spec/core/binary/index.html
+* https://github.com/WebAssembly/tool-conventions/blob/main/DynamicLinking.md
```

Comment:
Is this an accurate rephrasing?
```suggestion
Emscripten is a POSIX-based platform. It uses the `WebAssembly binary format`_,
including the `dynamic linking extension`_.

.. _WebAssembly binary format: https://webassembly.github.io/spec/core/binary/index.html
.. _dynamic linking extension: https://github.com/WebAssembly/tool-conventions/blob/main/DynamicLinking.md
```

Reply:
Yes, thanks. I will replace the word "extension" with the more specific word "section".